### PR TITLE
Cleanup of class FrameChangeObserver

### PR DESCRIPTION
**Pull request by DanRathbun**
See Issue #9
Correction of mistakes, and edit of examples.

### CODE REVIEW

**Comment by jimfoltz** [comment]
The first line:

    This observer interface is implemented to react to changes in camera position 

Frame changes are not related to the camera position.  Scenes may or may not include the camera position property, which is just one of several properties associated with a Scene.

**Comment by DanRathbun** [comment]
1. This is not my words. (It has said this forever.)
2. It is true that you can have two scenes that have exactly the same camera position, and animating from one to the next wouldn't appear to move the camera at all.
3. Camera (eye) position is not the only thing that is animated. The up vector and target are also animated.
4. Parts of the scene's styles are also animated. My standard template has a dark theme for the "Work" scene, and white backgrounds for the "Print" and "Thumbnail" scenes. The background color is animated (gradually changed) from one to the other. 
5. Other style settings (Edge, Face and Background) all switch on/off one the first frame. 
6.. The Modeling Settings are a mix. The various object colors (ie, Guides, etc.,) animate between scenes. But the toggles (like Axes, Hidden Geometry, Back Edges, etc.) all switch on the first frame.

I believe the introduction is "trying" to speak of "why would" the plugin programmer _basically_ create an observer of the class. **Not** why was this observer class invented for the SketchUp API. In that light, it may not use the best choice of grammar.

Do you have a better suggestion ?

Are there better examples of "summary" lines in other classes ? (Keep in mind that this first line, appears by itself in the summary section.)

And can we use this scenario to make a "rule" for the guidelines ?

**Comment by jimfoltz** [comment]
> Do you have a better suggestion ?

Just take the sentence out. 

    This observer's only callback method is called by SketchUp when the user makes a scene change, a Ruby script makes a scene change, or the internal animation feature runs.

**Comment by DanRathbun** [comment]
> Just take the sentence out.

 _I meant a better suggestion for the summary line._ 

And that suggestion is a bit vague. The callback is called on each change of frame, not by what your suggested sentence says.

THIS is OFFTOPIC of this PR. The PR was to fix what it said (the order of parameters and examples.)
Commentary for this PR, should concentrate on this PR and the open Issue it wants to satisfy.

**The content overhaul of the docstring should be opened as a new issue**.

(Ie, most users new to GitHub will not know that there are commentary threads attached to the bottom of PRs and Commits. Others should have the ease of finding and participating in this discussion regarding the class docstring.)

So, if you open an new issue (please,) I'll copy my comment above (with the item list) to the new issue.

File: SketchUp/Sketchup/FrameChangeObserver.rb [hunk 1]
```diff
@@ -36,13 +36,23 @@
...
+#           puts "From page '#{from_scene.name}' to '#{to_scene.name}'"
+#         end
+#       else
+#         Sketchup.status_text= "% done: #{'|'*(percent_done*100).to_i}"
+#         UI.messagebox("All Done!") if percent_done >= 1.0
+#       end
 #     end
 #   end
-#   id = Sketchup::Pages.add_frame_change_observer(MyFrameChangeObserver.new)
+#   @id = Sketchup::Pages.add_frame_change_observer(MyFrameChangeObserver.new)
```
Comment (thomthom) [inline SketchUp/Sketchup/FrameChangeObserver.rb]: Why the `@`?

**Comment by thomthom** [comment]
I'm ok with merging as-is. Though I'm curious of the instance variable change?

File: SketchUp/Sketchup/FrameChangeObserver.rb (hunk 1 above)
Comment (DanRathbun) [inline SketchUp/Sketchup/FrameChangeObserver.rb]: Best practice. You need a persistent reference so as to detach it later.

**Comment by DanRathbun** [comment]
As I said above (in the codebox, whatever it is ?):

Best practice. You need a persistent reference so as to detach it later.

**Pull request closed without merging**